deepnote_toolkit/sql/sql_execution.py

```
@@ -361,40 +362,53 @@ def _handle_federated_auth_params(sql_alchemy_dict: dict[str, Any]) -> None:
 
+def _open_ssh_tunnel(sql_alchemy_dict) -> tuple[Any, Any]:
+    """Open an SSH tunnel for *sql_alchemy_dict* and return ``(server, url)``.
+
+    Caller is responsible for eventually closing the returned server with
+    ``_close_ssh_tunnel``.
+    """
+    base64_encoded_key = dnenv.get_env("PRIVATE_SSH_KEY_BLOB")
+    if not base64_encoded_key:
+        raise Exception(
+            "The private key needed to establish the SSH connection is missing. Please try again or contact support."
+        )
```

Ruff (0.15.10)

[warning] Create your own exception

(TRY002)